controllers/machinehealthcheck_controller.go

```diff
@@ -58,6 +58,12 @@ const (
 	EventRemediationRestricted string = "RemediationRestricted"
 )
 
+// +kubebuilder:rbac:groups=core,resources=events,verbs=get;list;watch;create;patch
+// +kubebuilder:rbac:groups=core,resources=secrets,verbs=get;list;watch
+// +kubebuilder:rbac:groups=core,resources=nodes,verbs=get;list;watch;create;update;patch;delete
```

Comment:
I'm not sure if this is necessary or not, technically it needs the ability to read the nodes in the remote cluster, I think that's about it, do remote cluster permission live here? 🤔

Reply:
These permissions only apply to the management cluster, CAPI has admin access to workload cluster through the kubeconfig secret

Reply:
Thanks, I'm gonna go through and filter these down a bit, I think they're too wide for what the controller actually needs right now :)